src/adapter/src/catalog/builtin.rs

```diff
@@ -2371,6 +2371,67 @@ JOIN mz_catalog.mz_roles role_owner ON role_owner.id = class_objects.owner_id
 WHERE mz_schemas.database_id IS NULL OR d.name = pg_catalog.current_database()",
 };
 
+pub const PG_DEPEND: BuiltinView = BuiltinView {
+    name: "pg_depend",
+    schema: PG_CATALOG_SCHEMA,
+    sql: "CREATE VIEW pg_catalog.pg_depend AS WITH class_objects AS (
```

Comment:
I'm not super familiar with `pg_depend`, but I'm assuming that you're purposely leaving out all other object types that would be found in `mz_objects` but not `mz_relations` or `mz_indexes`. Specifically, `sink`s, `type`s, `connection`s, `function`s, and `secrets`. Those all have dependencies to consider when performing a drop but are not included here.

Can you also add some tests that check the contents of this view?

Reply:
Yes, that was on purpose. The reason is those objects are not exposed via pg_catalog and there's no relation I could reference for the `classid` column. I agree it's not ideal. 

And yes, I can add something.

Reply:
Linking to #9204, which @ParkMyCar had started looking into.

Reply:
> Linking to https://github.com/MaterializeInc/materialize/issues/9204, which @ParkMyCar had started looking into.

Good spot. I added "fixes #9204" to the PR description.